### ISSUE TITLE
`unresolved external` linking x86 WinDivert32.lib on x86_64 system

### ISSUE THREAD

**Issue by PantherJohn** [basil00/Divert#149]
While `WinDivert.lib` works fine here.

**Comment by PantherJohn** [comment]
```
#ifdef _WIN64
#pragma comment(lib, "WinDivert.lib")
#else
#pragma comment(lib, "WinDivert32.lib")
#endif
```
When I compiled the solution with option `x86 Release` `WinDivert32.lib` was not used.

**Comment by basil00** [comment]
I think this is a compiler problem, not a WinDivert problem?  Maybe there is a VS expert who can help.

**Comment by PantherJohn** [comment]
@basil00 the problem reminded me of the buggy nature of Windows -- I've heard they moved `Linker->Additional Dependencies` to some creepy place and left the original one invalid.

**Comment by TechnikEmpire** [comment]
@PantherJohn There's absolutely nothing wrong with visual studio 2017, you're just doing it wrong. Look I love to beat up on companies like Microsoft but really they're doing nothing wrong in these departments lately. I once paid over $800 for a lower level copy of visual studio so I'm thankful to have it for $0 these days.

You have not shared any information at all. Just "it doesn't work." Or "it didn't use this library." Well, there's several thousands of things that could have gone wrong, so people need error codes to help you. Can you please post the actual error codes?

@basil00 Would you accept a PR that has visual studio projects for both the driver and the dll?

**Comment by basil00** [comment]
@TechnikEmpire Yes a PR would be welcome.  I'd also prefer a command-line build script too if possible.

**Comment by PantherJohn** [comment]
@basil00 @TechnikEmpire 
I know it's a bit annoying if I don't paste error messages here. The output window said 
```
1>eLitexMain.obj : error LNK2001: unresolved external symbol __imp__WinDivertOpen
1>eLitexMain.obj : error LNK2001: unresolved external symbol __imp__WinDivertRecv
1>eLitexMain.obj : error LNK2001: unresolved external symbol __imp__WinDivertHelperParsePacket
1>eLitexMain.obj : error LNK2001: unresolved external symbol __imp__WinDivertHelperCalcChecksums
1>eLitexMain.obj : error LNK2001: unresolved external symbol __imp__WinDivertSend
1>C:\Users\RexJo\source\repos\eLitetX\Release\eLitetX.exe : fatal error LNK1120: 5 unresolved externals
```

**Comment by basil00** [comment]
All I can say is that `__imp__WinDivertOpen`, etc., are the x86 (32bit) symbol names.  So probably cannot find the 32bit DLL?

**Comment by PantherJohn** [comment]
@basil00 I use lib linking and the x64 project was not linked against x64 DLL either. I may try though. So what does that that mean? the precompiled lib doesn’t seem to support static linking. I compiled the x64 one and it still required me to put WinDivert.dll in the directory for it to work.

**Comment by PantherJohn** [comment]
#121

**Comment by basil00** [comment]
The `WinDivert.lib` is a *DLL import library* and not a static library, which may cause confusion.  You still need the `WinDivert.dll` file.  There is a 32bit and 64bit version of the `WinDivert.dll` file, and you must link against the correct one.

**Comment by TechnikEmpire** [comment]
Theres nothing wrong here except a misunderstanding of what's happening. First, be in a dll or a static lib, msvc only cares about the lib file. The lib file defines the exports both in shared and static linking. No dlls required.

If you want static linking then you must compile a static lib. In the "Linker" or "Librarian" properties (depends on what kind of project you're building) you need to include both directories for both lib files. In fact if you're doing static linking you might as well make the output file name for both architectures the same name and then msvc will select the correct lib based on it's matching arch and you can dump the whole preprocessor lib declaration.

**Comment by PantherJohn** [comment]
@basil00 @TechnikEmpire 
> The WinDivert.lib is a DLL import library and not a static library
> The lib file defines the exports both in shared and static linking
gotcha thanks you guys

**Comment by Barbod69** [comment]
use this
https://copyfuture.com/blogs-details/202204051455480328
